Sources/Build/BuildPlan.swift

```diff
@@ -744,7 +744,7 @@ public final class SwiftTargetBuildDescription {
 
     public static func checkSupportedFrontendFlags(flags: Set<String>, fileSystem: FileSystem) -> Bool {
         do {
-            let executor = try SPMSwiftDriverExecutor(resolver: ArgsResolver(fileSystem: fileSystem), fileSystem: fileSystem, env: [:])
+            let executor = try SPMSwiftDriverExecutor(resolver: ArgsResolver(fileSystem: localFileSystem), fileSystem: fileSystem, env: [:])
```

Comment:
This was wrong in the existing code.  Much as we'd want to use the file system abstraction everywhere, the `swift` compiler assumes access to the local file system and can't access the `InMemoryFileSystem` used in unit tests (or any other virtual file system either for that matter).

Reply:
what does arg resolver do that depends on hard coding the local file system?

Reply:
Passing it to the ArgsResolver is a mistake actually — I meant to pass that to the SPMSwiftDriverExecutor.  Which makes it extra odd that that fixed the issue.

It looks as if it creates a file in the local file system and invokes the Swift compiler.  So my comments were based on the fact that the compiler deals with the real file system and nothing else.  That also makes this fairly expensive — I was trying to keep this a small PR but really this should be reworked to deal much better with looking up supported features — I'll see what I can do that doesn't turn into a side project here.

Reply:
Removed this again as the caller can pass localFileSystem (as indeed all the other call sites do)